### PR TITLE
transact endpoint to POST /transact  GET /transact and GET /transact/:id added

### PR DESCRIPTION
**Pull request by ghost**
Endpoints to save get all and get by id added

POST /transact - Save transactions in 4 kinds (Transactions, Debitor, Creditor, TransactionItems) - Transaction is parent of all debitor, creditor and transaction item

GET /transact - It ill list all entities, Didn't changed response 
GET /transact/:id  - It ill list entity of given Id in req.params.id, Didn't changed response.

### CODE REVIEW

**Comment by mxfactorial** [comment]
Please add example requests & responses as codeblocks or screenshots for client developers, similar to https://github.com/systemaccounting/mxfactorial/issues/24. Minimizes guesswork.

**Comment by ghost** [comment]
@mxfactorial 

Sure thing - I am going to add it in few minutes.

**Comment by ghost** [comment]
Here are screen shots of POSTMan showing calls for POST, GET for all records and details of transaction

POST /transact to save new transaction
Headers : Content-Type application/json  
image

Sample of post JOSN in body 

```
{
    "debitor":"John",
    "creditor":"David",
    "db_time":"10:01",
    "db_latlng":"40.7128,74.0059",
    "cr_time":"10:02",
    "cr_latlng":"40.7128,74.0059",
    "transaction_item" : [
        {"value":10,
         "name":"Milk",
         "quantity":1,
         "units_measured":"1",
         "unit_of_measurement":"Ltr"
        },{"value":49,
         "name":"Bread",
         "quantity":3,
         "units_measured":"1",
         "unit_of_measurement":"Packs"
        }
    ]
}
```

It ill response back with transaction id - Auto generated

GET /transact/:id - Provide id of transaction to get specific transaction
image

GET /transact  List all transaction
image

File: transact/crud.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// Automatically parse request body as form data
+router.use(bodyParser.urlencoded({ extended: false }));
+router.use(bodyParser.json({type:'application/json', limit:'500mb'})); 
+
+//save transaction
+router.post('/', function (req, res) {
+  var body = req.body;
+
+  console.log(body);
+  if (!body.transaction_item || body.transaction_item<=0) {
```
Comment (hotay) [inline transact/crud.js]: duplicated, and did you mean `body.transaction_item.length` here?

**Comment by mxfactorial** [comment]
After adding .length, a 400 error is no longer received, but the Datastore does not receive the object. Also, web server prints an empty object upon receiving post request:
empty-object

**Comment by ghost** [comment]
@mxfactorial pass content type application json in header.

File: transact/crud.js (hunk 1 above)
Comment (ghost) [inline transact/crud.js]: ops, Yes i mean .length. My mistake

**Comment by mxfactorial** [comment]
Passes testing.

https://[BASE_URL]/systemaccounting/transact
Header key: `Content-Type`
Header value: `application/json`
Body:

```
{
    "db_author":"John",
    "cr_author":"David",
    "db_time":"10:01",
    "db_latlng":"40.7128,74.0059",
    "cr_time":"10:02",
    "cr_latlng":"40.7128,74.0059",
    "transaction_item" : [
        {"db_account":"John",
         "cr_account":"David",
         "quantity":1,
         "units_measured":"1",
         "unit_of_measurement":"Ltr",
         "value":10,
         "name":"Milk"
        },{"db_account":"John",
         "cr_account":"David",
         "quantity":1,
         "units_measured":"1",
         "unit_of_measurement":"Ltr",
         "value":5,
         "name":"Bread"
        }
    ]
}
```

**Pull request merged**